--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -119,7 +119,7 @@ set(Python3_FIND_REGISTRY LAST)
 if(BUILD_SHARED_LIBS)
   set(Python3_USE_STATIC_LIBS FALSE)
 endif()
-find_package_auto(ENABLE_PYTHON_SCRIPTING Python3 3.3 COMPONENTS Development Interpreter)
+find_package_auto(ENABLE_PYTHON_SCRIPTING Python3 3.3 COMPONENTS Development.Module Development.Embed Interpreter)
 find_package_auto(ENABLE_LIBSPIRO         Libspiro)
 find_package_auto(ENABLE_LIBUNINAMESLIST  Libuninameslist)
 
--- a/fontforge/CMakeLists.txt
+++ b/fontforge/CMakeLists.txt
@@ -253,7 +253,7 @@ if(ENABLE_LIBREADLINE_RESULT)
   target_link_libraries(fontforge PUBLIC Readline::Readline)
 endif()
 if(ENABLE_PYTHON_SCRIPTING_RESULT)
-  target_link_libraries(fontforge PUBLIC Python3::Python)
+  target_link_libraries(fontforge PUBLIC Python3::Module)
 endif()
 if(ENABLE_WOFF2_RESULT)
   target_link_libraries(fontforge PRIVATE WOFF2::WOFF2)
--- a/fontforgeexe/CMakeLists.txt
+++ b/fontforgeexe/CMakeLists.txt
@@ -105,6 +105,9 @@ else()
 endif()
 
 target_link_libraries(fontforgeexe PRIVATE fontforge Iconv::Iconv MathLib::MathLib)
+if(ENABLE_PYTHON_SCRIPTING_RESULT)
+  target_link_libraries(fontforgeexe PRIVATE Python3::Python)
+endif()
 install(TARGETS fontforgeexe RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 
 if(UNIX)
